### DISCUSSION TITLE
Tasklib not found when installing with Vim-plug

### DISCUSSION THREAD

**Discussion by fullstopslash** [tbabej/taskwiki#329] [Community support]
I was catching this error on Mac Big Sur. No matter where I seemed to install tasklib. 
```
Error detected while processing function provider#python3#Call:        
line   18:
Error invoking 'python_execute_file' on channel 4 (python3-script-host):
Traceback (most recent call last):
  File "/Users/danielwilcox/.vim/plugged/taskwiki/taskwiki/main.py", line 15, in
 <module>
    from taskwiki import cache as cache_module
  File "/Users/danielwilcox/.vim/plugged/taskwiki/taskwiki/cache.py", line 5, in
 <module>
    from taskwiki import preset
  File "/Users/danielwilcox/.vim/plugged/taskwiki/taskwiki/preset.py", line 5, i
n <module>
    from taskwiki import util
  File "/Users/danielwilcox/.vim/plugged/taskwiki/taskwiki/util.py", line 11, in
 <module>
    import tasklib
ModuleNotFoundError: No module named 'tasklib'
Press ENTER or type command to continue
```
It turns out, as I'm using [Vim-Plug](https://github.com/junegunn/vim-plug) to manage my plugins I guess it does some kind of sandboxing for each plugin. 

The solution to get it working was to copy the libraries for pytz, six, tzlocal, and tasklib, to the plugin's directory for python. In my case ```~/.vim/plugged/taskwiki/pythonx/```   I hope this helps anyone who runs into this in the future!

**Comment by tbabej** [comment]
Vim-plug does not do any sandboxing on its own, but I think the issue is that the system python and the python available from vim on Mac OSX use different locations for site-packages.

**Comment by poetaman** [comment]
In my case I would get the same error on my system with vim, but not with neovim.

```
Mac OS X	Homebrew: brew install task 
Homebrew: brew install taskd 
Homebrew: brew install tasksh 
```

I got those install steps from https://taskwarrior.org/download/.

Doing ``pip[3] install tasklib`` makes taskwiki work in both vim and neovim. I totally missed doing that as somehow neovim was able source tasklib...
